pkg/stanza/operator/input/gelfinternal/config.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gelfinternal // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/udp"
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+)
+
+const (
+	operatorType = "gelf_input"
+	defaultReaders           = 1
+	defaultProcessors        = 1
+	defaultUDPMaxQueueLength = 100
+	defaultListenAddress     = "0.0.0.0:31250"
+	defaultProtocol          = "udp"
+	MaxUDPSize               = 64 * 1024
+)
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+// NewConfig creates a new UDP input config with default values
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfigWithID creates a new UDP input config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		InputConfig: helper.NewInputConfig(operatorID, operatorType),
+		BaseConfig: BaseConfig{
+			ListenAddress:     string(defaultListenAddress),
+			Protocol:          string(defaultProtocol),
+			AsyncReaders:      defaultReaders,
+			AsyncProcessors:   defaultProcessors,
+			UDPMaxQueueLength: defaultUDPMaxQueueLength,
+		},
+	}
+}
+
+// Config is the configuration of a udp input operator.
+type Config struct {
+	helper.InputConfig `mapstructure:",squash"`
+	BaseConfig         `mapstructure:",squash"`
+}
+
+// BaseConfig is the details configuration of a udp input operator.
+type BaseConfig struct {
+	ListenAddress     string `mapstructure:"listen_address,omitempty"`
+	Protocol          string `mapstructure:"protocol,omitempty"`
+	AsyncReaders      int    `mapstructure:"async_readers,omitempty"`
+	AsyncProcessors   int    `mapstructure:"async_processors,omitempty"`
+	UDPMaxQueueLength int    `mapstructure:"udp_max_queue_length,omitempty"`
+}
+
+// Build will build a udp input operator.
+func (c Config) Build(set component.TelemetrySettings) (operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(set)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.ListenAddress == "" {
+		return nil, fmt.Errorf("missing required parameter 'listen_address'")
+	}
+
+	if _, _, err := net.SplitHostPort(c.ListenAddress); err != nil {
+		return nil, fmt.Errorf("invalid listen_address: %w", err)
+	}
+
+	if c.Protocol != "udp" {
+		return nil, fmt.Errorf("supported protocols - udp, invalid protocol: %s", c.Protocol)
+	}
+	if c.AsyncReaders <= 0 && c.AsyncReaders < 20 {
```

Comment:
are you sure of these conditions?

Reply:
I am not sure, just wanted to add some boundary conditions, I'll remove it.